e2e/testcases/private_cert_secret_test.go

```diff
@@ -280,6 +280,25 @@ func TestCACertSecretRefV1Beta1(t *testing.T) {
 	if err != nil {
 		nt.T.Fatal(err)
 	}
+
+	// Set RepoSync to use HTTPS with no auth and SSL verification disabled.
+	// This validates that the GIT_SSL_CAINFO env var can be removed.
```

Comment:
I believe line 266 should already be verifying this. It unsets `caCertSecretRef` and then checks that the env var has been removed

Reply:
Technically you're correct, but it also doesn't sync. I wanted one where it actually worked.

Reply:
Seems like you could add a `DeploymentMissingEnvVar` predicate to line 279 then, right?

Reply:
Fair enough.

That said, I don't understand how 278 works without the CACertSecretRef. The SecretRef is a private key to authenticate the user, right? How does it validate the server cert without a CA file?

Reply:
Does git@ scheme not validate the server cert by default??

Reply:
caCertSecretRef is only used for verifying the authenticity of the server certificate over HTTPS. SSH is a different protocol

Reply:
Does SSH not validate the server cert? I thought it did.

Or is the git server configured to use a known CA when using SSH?

Reply:
Certificate authorities aren't involved with SSH protocol. They are used to sign certificates used in HTTPS. Authenticity over SSH is established using the public/private key pair

Reply:
> In the connection setup phase, the SSH server authenticates itself to the client by providing its public key. This allows the SSH client to verify that it is actually communicating with the correct SSH server (instead of an attacker that could be posing as the server).

Weird... How can you trust that the SSH server is who they say they are if you get the public key from them instead of a trusted third party? In TLS, the server gives you its public key, but it has to have been signed by a CA you trust. It sounds like maybe SSH (or at least git?) doesn't have that layer of security by default, which is... really surprising.